repos/system_upgrade/common/actors/scanthirdpartytargetpythonmodules/libraries/scanthirdpartytargetpythonmodules.py

```diff
@@ -0,0 +1,189 @@
+import json
+import os
+from collections import defaultdict
+from pathlib import Path
+
+import rpm
+
+from leapp.libraries.common.config.version import get_target_major_version
+from leapp.libraries.common.rpms import has_package
+from leapp.libraries.stdlib import api, run
+from leapp.models import DistributionSignedRPM, ThirdPartyTargetPythonModules
+
+PYTHON_EXTENSIONS = (".py", ".so", ".pyc")
+FMT_LIST_SEPARATOR = '\n  - '
+
+
+def _formatted_list_output(input_list, sep=FMT_LIST_SEPARATOR):
+    return ['{}{}'.format(sep, item) for item in input_list]
+
+
+def get_python_sys_paths(python_interpreter):
+    """Get sys.path from the specified Python interpreter."""
+
+    result = run([python_interpreter, '-c', 'import sys, json; print(json.dumps(sys.path))'])['stdout']
+    return json.loads(result)
```

Comment:
You are frequently ad-hoc constructing a `Path` object and then throwing it away (when resolving symlinks, when using `rglob`). Why not construct it once, and use it where required? 
```suggestion
    result = run([python_interpreter, '-c', 'import sys, json; print(json.dumps(sys.path))'])['stdout']
    raw_paths = json.loads(result)
    paths = [Path(raw_path) for raw_path in raw_paths]
    return paths
```